### PR TITLE
add getDataDir to std/appdirs.nim

### PR DESCRIPTION
**Pull request by Ecorous**
This PR extends #21408 by adding the `getDataDir` proc to the file `lib/std/appdirs.nim`

### CODE REVIEW

**Comment by Araq** [comment]
CI is failing for some reason but appart from that it is acceptable.

**Comment by Ecorous** [comment]
> CI is failing for some reason but appart from that it is acceptable.

It did that on #21408 as well. Not sure why it did. Never ended up fixing it

**Comment by ghost** [comment]
In this PR the error seems to actually be relevant - 
image

The `/` proc for string + string is defined in [`std/os`](https://nim-lang.org/docs/os.html#%2F%2Cstring%2Cstring) not in `std/paths`

Also, aren't rtl and extern wrong here? Other procs in the same std/appdirs module don't use those, and the code doesn't work with them for me anyway. Have you tested this locally?

**Comment by ghost** [comment]
Wait, at this point I'm really confused. Why do we have https://github.com/nim-lang/Nim/blob/devel/lib/std/private/osappdirs.nim and https://github.com/nim-lang/Nim/blob/devel/lib/std/appdirs.nim that both implement exactly the same things? @ringabout you were involved with them, can you please explain to me? I'm probably just missing some context, but this does seem quite confusing, especially because the modules reimplement the same code and not just import the other one.

**Comment by ringabout** [comment]
@Yardanico because `std/appdirs` introduces `Path` interface, which introduces breaking changes. `std/private/osappdirs` is used for std/os. I don't merge these two modules to keep backwards compatibility.

File: lib/std/appdirs.nim [hunk 1]
```diff
@@ -17,6 +17,30 @@ proc getHomeDir*(): Path {.inline, tags: [ReadEnvEffect, ReadIOEffect].} =
...
+  ## * `expandTilde proc`_
+  ## * `getCurrentDir proc`_
+  ## * `setCurrentDir proc`_
+  when defined(windows):
+    result = getEnv("APPDATA")
+  elif defined(macosx):
+    result = getEnv("XDG_DATA_HOME", getEnv("HOME") / "Library" / "Application Support")
+  else:
+    result = getEnv("XDG_DATA_HOME", getEnv("HOME") / ".local" / "share")
+  result.normalizePathEnd(trailingSep = true)
```
Comment (ringabout) [inline lib/std/appdirs.nim]:
```suggestion
  result = Path(osappdirs.getDataDir())
```

**Review by ringabout — CHANGES_REQUESTED** [review]

File: lib/std/appdirs.nim (hunk 1 above)
Comment (ringabout) [reply lib/std/appdirs.nim]: Please reuse the implementation of https://github.com/nim-lang/Nim/blob/fe98cea9cff03cfbb32374208e83b9817694bf1c/lib/std/private/osappdirs.nim#L28

File: lib/std/appdirs.nim [hunk 2]
```diff
@@ -17,6 +17,24 @@ proc getHomeDir*(): Path {.inline, tags: [ReadEnvEffect, ReadIOEffect].} =
   ## * `getTempDir proc`_
   result = Path(osappdirs.getHomeDir())
 
+proc getDataDir*(): string {.rtl, extern: "nos$1"
+  tags: [ReadEnvEffect, ReadIOEffect].} =
```
Comment (ringabout) [inline lib/std/appdirs.nim]:
```suggestion
proc getDataDir*(): Path {.inline, tags: [ReadEnvEffect, ReadIOEffect].} =
```

Comment (ringabout) [inline lib/std/appdirs.nim]:
```suggestion
proc getDataDir*(): Path {.inline, tags: [ReadEnvEffect, ReadIOEffect].} =
```

**Review by ringabout — APPROVED** [review]
LGTM. Since it's a new module, it doesn't need a changelog entry for a single proc.

**Pull request merged**